main.nf

```diff
@@ -191,67 +235,655 @@ process get_software_versions {
+  msgfinstrument = [velos:1, qe:3, false:0][params.instrument]
+  """
+  msgf_plus -Xmx16G -d $db -s $x -o "${sample}.mzid" -thread 12 -mod $mods -tda 0 -t 10.0ppm -ti -1,2 -m 0 -inst ${msgfinstrument} -e 1 -protocol ${msgfprotocol} -ntt 2 -minLength 7 -maxLength 50 -minCharge 2 -maxCharge 6 -n 1 -addFeatures 1
+  msgf_plus -Xmx3500M edu.ucsd.msjava.ui.MzIDToTsv -i "${sample}.mzid" -o out.mzid.tsv
+  rm ${db.baseName.replaceFirst(/\.fasta/, "")}.c*
+  """
+}
+
+mzids
+  .groupTuple()
+  .set { mzids_2pin }
+
+
+process percolator {
+
+  input:
+  set val(setname), val(samples), file('mzid?') from mzids_2pin
+
+  output:
+  set val(setname), file('perco.xml') into percolated
+
+  """
+  echo $samples
+  mkdir mzids
+  count=1;for sam in ${samples.join(' ')}; do ln -s `pwd`/mzid\$count mzids/\${sam}.mzid; echo mzids/\${sam}.mzid >> metafile; ((count++));done
+  msgf2pin -o percoin.xml -e trypsin -P "decoy_" metafile
+  percolator -j percoin.xml -X perco.xml -N 500000 --decoy-xml-output -y
+  """
+}
+
+
+mzidtsvs
+  .groupTuple()
+  .join(percolated)
+  .set { mzperco }
+
+
+process svmToTSV {
+
+  input:
+  set val(setname), file('mzident????'), file('mzidtsv????'), val(platenames), val(fractions), file(perco) from mzperco 
+
+  output:
+  set val(setname), val('target'), file('tmzidperco') into tmzidtsv_perco
+  set val(setname), val('decoy'), file('dmzidperco') into dmzidtsv_perco
+
+  script:
+  """
+  perco_to_tsv.py -p $perco --plates ${platenames.join(' ')} --fractions ${fractions.join(' ')}
+  """
+}
+
+// Collect percolator data of target/decoy and feed into PSM table creation
+tmzidtsv_perco
+  .concat(dmzidtsv_perco)
+  .groupTuple(by: 1)
+  .combine(quant_lookup)
+  .set { prepsm }
+
+// Set strips to false if not running hirief
+if (params.hirief) {
+  strips_for_deltapi
+    .map { it -> [it, trainingpep] }
+    .set { stripannot }
+} else {
+  strips_for_deltapi
+    .map { it -> [it, false, false]}
+    .set { stripannot }
+}
+
+
+/*
+* Step 3: Post-process peptide identification data
+*/
+
+process createPSMTable {
+
+  publishDir "${params.outdir}", mode: 'copy', overwrite: true, saveAs: {["target_psmlookup.sql", "target_psmtable.txt", "decoy_psmtable.txt"].contains(it) ? it : null}
+
+  input:
+  set val(setnames), val(td), file('psms?'), file('lookup') from prepsm
+  set file(tdb), file(ddb) from searchdbs
+  set val(allstrips), file(trainingpep) from stripannot
+
+  output:
+  set val(td), file("${td}_psmtable.txt") into psm_result
+  set val(td), file({setnames.collect() { it + '.tsv' }}) into setpsmtables
+  set val(td), file("${td}_psmlookup.sql") into psmlookup
+
+  script:
+  """
+  msspsmtable merge -i psms* -o psms.txt
+  msspsmtable conffilt -i psms.txt -o filtpsm --confidence-better lower --confidence-lvl 0.01 --confcolpattern 'PSM q-value'
+  msspsmtable conffilt -i filtpsm -o filtpep --confidence-better lower --confidence-lvl 0.01 --confcolpattern 'peptide q-value'
+  cp lookup psmlookup
```

Comment:
`cat lookup > psmlookup` is more explanatory for users checking this code here 👍 

If you don't need the file as an extra one - just use $lookup in the next steps...